src/mui/detail/Edit.js

```diff
@@ -30,7 +31,11 @@ export class Edit extends Component {
 
     componentWillReceiveProps(nextProps) {
         if (this.props.data !== nextProps.data) {
-            this.setState({ record: nextProps.data }); // FIXME: erases user entry when fetch response arrives late
+            this.setState({ record: nextProps.data });
+
+            // As a refresh of the current record won't actually change the record values,
+            // we need to explicitly ask redux-form to reset the form
+            this.props.resetForm('record-form');
             if (this.fullRefresh) {
```

Comment:
the `fullRefresh` thingie is never updated, I think it's a trace from the past (when we didn't use the store for refreshes). 

We need another way to force a redraw.